### DISCUSSION TITLE
RAD2 modpack running with docker-compose not able to locate ZIP.

### DISCUSSION THREAD

**Discussion by Kingrune5550** [itzg/docker-minecraft-server#2098] [Q&A]
Hey there!

I am trying to deploy a container with RAD2(https://www.curseforge.com/minecraft/modpacks/roguelike-adventures-and-dungeons-2) modpack using docker-compose. I ran into some issues:

Docker Compose file for reference: 

```json
version: "3"

services:
  rad2vrc:
    image: itzg/minecraft-server:java8-jdk
    container_name: minecraftserver
    restart: unless-stopped
    user: 1000:1000
    security_opt:
      - no-new-privileges:true
    sysctls:
      - net.ipv4.ip_unprivileged_port_start=0
    networks:
      - mc
    ports:
      - 25565:25565
    volumes:
      - ./minecraft-data:/data
    environment:
      EULA: "TRUE"
      TYPE: "AUTO_CURSEFORGE"
      VERSION: "1.16.5"
      MEMORY: "10G"
      CF_API_KEY: "$$2a$$10$$jUDbtvPnFeC2O4ezHjNOEuUd14Ty.k54OxCdCRyKdgaUa6cMoV54W"
      CF_PAGE_URL: "https://www.curseforge.com/minecraft/modpacks/roguelike-adventures-and-dungeons-2/files/4488131"
      CF_MODPACK_ZIP: "./RAD2-1.16.5-1.2.zip"
      CF_EXCLUDE_MODS: "itemphysic-lite,craftpresence,item-borders,betterf3"
      MOTD: "The Holonet MC Server"
      DIFFICULTY: "normal"
      ENABLE_WHITELIST: "TRUE"
      ENFORCE_WHITELIST: "TRUE"
      MAX_PLAYERS: "10"
      ALLOW_NETHER: "true"
      SNOOPER_ENABLED: "false"
      MODE: "survival"
      FORCE_GAMEMODE: "true"
      ENABLE_ROLLING_LOGS: "true"
      TZ: "America/Toronto"
      MAX_TICK_TIME: "-1"
      JVM_DD_OPTS: "disable.watchdog:true"
      ENABLE_AUTOPAUSE: "TRUE"
      AUTOPAUSE_TIMEOUT_EST: "600"
      AUTOPAUSE_TIMEOUT_INIT: "600"
      AUTOPAUSE_TIMEOUT_KN: "30"
    tty: true
    stdin_open: true

networks:
  mc:
    external: true
```

- These three mods are disallowed for project distribution:
	- BetterDefaultBiomes
	- Magical Jewelry
	- Harder Branch Mining
I saw that there is the line: "If the authors of the modpack have disallowed project distribution, then the desired client modpack zip will need to be manually downloaded and made available to the container. The path to that file must be passed to CF_MODPACK_ZIP." I assume since only some mods in the modpack are marked disallowed project distribution, then I would still need to specify the file.
I downloaded the client file as specified by the above line but get: "java.nio.file.NoSuchFileException: ./RAD2-1.16.5-1.2.zip" but the zip file is in the same directory as the docker-compose shown here: 
![image](https://user-images.githubusercontent.com/18042763/234063396-b1c87baf-54c7-4d0d-a0ca-1947e2095782.png)

- Second is related to the CF_EXCLUDE_MODS section. all of the mods there are client side only except for betterf3 (As I don;t see it listed on the mod page). I tracked down the fileID (3293637) of the mod to confirm if it was compatible with minecraft version 1.16.5 and it is. but after excluding all of these mods and running the server (without specifying the zip in the above step), when I login to the server I get forced removed. seeing as: 
![image](https://user-images.githubusercontent.com/18042763/234068295-949f6f00-22b9-455c-9a53-a6950b3a3151.png)
I believe this is due to removing the mods but not sure the best course of action since I believe the CF URL is pulling the client mod list which makes this incompatible?

Hopefully I explained myself clearly. let me know if you need any more information. 
Thanks!

**Comment by itzg** [comment]
> I downloaded the client file as specified by the above line but get: "java.nio.file.NoSuchFileException: ./RAD2-1.16.5-1.2.zip" but the zip file is in the same directory as the docker-compose

That is the wrong directory to put the zip into then. You have only mapped the subdirectory `./minecraft-data`. The container has no way to know/access the directory that contains the compose file without a mapping.

If you move the RAD2 zip into your `./minecraft-data`, then the relative path you attempted will work, since paths within the container are relative to `/data`.
